arangod/IResearch/IResearchFeature.cpp

```diff
@@ -294,6 +295,67 @@ uint32_t computeThreadsCount(uint32_t threads, uint32_t threadsLimit, uint32_t d
                            threads ? threads : uint32_t(arangodb::NumberOfCores::getValue()) / div));
 }
 
+bool upgradeArangoSearchLinkCollectionName(TRI_vocbase_t& vocbase,
+                                           arangodb::velocypack::Slice const& /*upgradeParams*/) {
+  using arangodb::application_features::ApplicationServer;
+  if (!arangodb::ServerState::instance()->isDBServer()) {
+    return true;  // not applicable for other ServerState roles
+  }
+  auto& selector = vocbase.server().getFeature<arangodb::EngineSelectorFeature>();
+  auto& clusterInfo =
+      vocbase.server().getFeature<arangodb::ClusterFeature>().clusterInfo();
+  // persist collection names in links
+  for (auto& collection : vocbase.collections(false)) {
+    auto indexes = collection->getIndexes();
+    auto clusterCollection =
+        clusterInfo.getCollectionNT(vocbase.name(),
+                                    std::to_string(collection->planId().id()));
+    if (clusterCollection) {
+      auto name = clusterCollection->name();
+      LOG_TOPIC("773b4", TRACE, arangodb::iresearch::TOPIC)
+          << " Processing collection " << name;
+#ifdef USE_ENTERPRISE
+      arangodb::iresearch::IResearchLinkHelper::realNameFromSmartName(name);
+#endif
+      for (auto& index : indexes) {
+        if (index->type() == arangodb::Index::IndexType::TRI_IDX_TYPE_IRESEARCH_LINK) {
+          auto indexPtr =
+              dynamic_cast<arangodb::iresearch::IResearchLink*>(index.get());
+          if (indexPtr) {
+            LOG_TOPIC("d6edb", TRACE, arangodb::iresearch::TOPIC)
+                << "Checking collection name '" << name << "' for link "
+                << indexPtr->id().id();
+            if (indexPtr->setCollectionName(name)) {
+              LOG_TOPIC("b269d", INFO, arangodb::iresearch::TOPIC)
+                  << "Setting collection name '" << name << "' for link "
+                  << indexPtr->id().id();
+              if (selector.engineName() == arangodb::RocksDBEngine::EngineName) {
+                auto& engine = selector.engine<arangodb::RocksDBEngine>();
+                auto builder =
+                  collection->toVelocyPackIgnore({"path", "statusString"},
+                                                 arangodb::LogicalDataSource::Serialization::PersistenceWithInProgress);
+                auto res =
+                    engine.writeCreateCollectionMarker(vocbase.id(), collection->id(),
+                                                       builder.slice(),
+                                                       arangodb::RocksDBLogValue::Empty());
```

Comment:
I believe we have to pass actual marker body

Reply:
We don't want to run recovery for that write. In case of failure during upgrade  - the upgrade will be started again and we will write the collection name eventually.